backend/services/implementations/userService.ts

```diff
@@ -16,6 +17,50 @@ const getMongoUserByAuthId = async (authId: string): Promise<User> => {
   return user;
 };
 
+/**
+ * Utility method for converting yyyy/mm/dd date
+ *  string to 24-char hex timestamp string
+ *
+ * ObjectID docs:
+ *  https://mongodb.github.io/node-mongodb-native/api-bson-generated/objectid.html
+ * Reference:
+ *  https://kchodorow.com/2011/12/20/querying-for-timestamps-using-objectids/
+ */
+const getDateTime = (date: string): string =>
+  `${Math.floor(new Date(date).getTime() / 1000).toString(16)}0000000000000000`;
+
+interface LimitByDateQuery {
+  $match: {
+    _id: {
+      [key: string]: Types.ObjectId;
+    };
+  };
+}
+
+const buildLimitByDateQuery = (
```

Comment:
Second thing on this, I think the logic for building queries should be separated from the user service, because again, it's pretty general and can be reused (i.e wanting to query courses or modules by date published). Only the top-level function (i.e buildLimitByDate) query **could** be specific to a given service. 

I think the best way to handle this is 
1. Create a queryBuilderService that can be reused across different services to create queries on the data model. 
2. To allow us to scope query builder functions only to services that need them, we should export objects of functions that contain different builder functions that each service can use. For example, if we have service A and B, which can only use builder function A and B respectively, but they can both use builder function C, then we'd export the following from the query builder service
``` 
const serviceAQueryBuilders = { queryBulderA, queryBuilderC }
const serviceBQueryBuilders = { queryBuilderB, queryBuilderC }
export default {serviceAQueryBuilders, serviceBQueryBuilders }
```
Then service A would import serviceAQueryBuilders and B would import B builders.